### DISCUSSION TITLE
Incorrect behaviour "SQLite doesn't support multiple calls to dropColumn / renameColumn in a single modification."

### DISCUSSION THREAD

**Discussion by Cryborg** [laravel/framework#50308] [Q&A] [answered]
### Laravel Version

10.46

### PHP Version

8.2

### Database Driver & Version

SQLite

### Description

In a migration, I have the following code that worked well a few days ago : 
```
    public function down(): void
    {
        Schema::table('answers', function(Blueprint $table) {
            $table->dropColumn('corrected_text');
            $table->dropColumn('options');
            $table->dropColumn('score');
        });
    }
```

Now, I have this message when I run the tests : "SQLite doesn't support multiple calls to dropColumn / renameColumn in a single modification."

The solution I found is this : 
```
    public function down(): void
    {
        Schema::table('answers', function (Blueprint $table) {
            $table->dropColumn('corrected_text');
        });
        Schema::table('answers', function (Blueprint $table) {
            $table->dropColumn('options');
        });
        Schema::table('answers', function (Blueprint $table) {
            $table->dropColumn('score');
        });
    }
```

But it seems hacky and looks like a regression to me. I updated to 10.46 and I was on 10.40 before.

### Steps To Reproduce

Use multiple dropColumns in a migration file, and run unit tests using SQLite DB.

**Comment by hafezdivandari** [comment, marked as answer]
This happens when you have `doctrine/dbal` package installed.

To fix this (Laravel >= 9 and < 11), you may call `Schema::useNativeSchemaOperationsIfPossible()` method within the boot method of your `App\Providers\AppServiceProvider` class. check #45258

PS: This won't happen on Laravel 11.x: #48864

**Comment by Cryborg** [comment]
> Schema::useNativeSchemaOperationsIfPossible()

It solved the problem, thanks!

Just for my knowledge, why do I have to do this now? doctrine/dbal has been installed for months on my project.

**Comment by hafezdivandari** [comment]
Nothing changed recently on this manner (dropping columns on 10.x), maybe you didn't run your migrations after installing doctrine/dbal?

**Reply by Cryborg** [reply]
I guess it has been added by one of the last packages we installed, because yes, migrations have been run recently. Thank you!

**Comment by marioxis** [comment]
I get the same error when I try to sync my DB. I use Laravel Framework 8.83.29 and PHP 7.4.33 (I know the PHP version is old, but my team is working on it). However, `Schema::useNativeSchemaOperationsIfPossible()` does not work because this feature was released for Laravel 9. Any suggestions? Thank you.

```
DatabaseMigrationSyncTest > migrations are synchronized
   PHPUnit\Framework\ExceptionWrapper 

  SQLite doesn't support multiple calls to dropColumn / renameColumn in a single modification.

  at vendor/laravel/framework/src/Illuminate/Database/Schema/Blueprint.php:156
    152▕     protected function ensureCommandsAreValid(Connection $connection)
    153▕     {
    154▕         if ($connection instanceof SQLiteConnection) {
    155▕             if ($this->commandsNamed(['dropColumn', 'renameColumn'])->count() > 1) {
  ➜ 156▕                 throw new BadMethodCallException(
    157▕                     "SQLite doesn't support multiple calls to dropColumn / renameColumn in a single modification."
    158▕                 );
    159▕             }
    160▕ 
```

**Reply by hafezdivandari** [reply]
On legacy versions (< 9), 
If you have mutiple drop columns, as I remember, Sqlite supports 1 command with multipe columns, like:

```php
$table->dropColumn(['col1', 'col2'])
```

But if you have mixture of commands, the only workaround is to use separate blueprints:

```php
Schema::table('test', function ($table) {
    $table->dropColumn(...);
});

Schema::table('test', function ($table) {
    $table->renameColumn(...);
});
```
